llm/ext_server/server.cpp

```diff
@@ -1382,114 +1424,161 @@ struct llama_server_context
         }
     }
 
-    void process_single_task(task_server& task)
+    std::string common_prefix(const std::string& str1, const std::string& str2) {
+        auto mismatch_pair = std::mismatch(str1.begin(), str1.end(), str2.begin());
+        return std::string(str1.begin(), mismatch_pair.first);
+    }
+
+    // Find the slot that has the greatest common prefix
+    server_slot *prefix_slot(const json &prompt) {
+        if (!prompt.is_string()) {
+            return nullptr;
+        }
+
+        std::string prompt_str = prompt.get<std::string>();
+        server_slot *slot = nullptr;
+        size_t longest = 1;
```

Comment:
Curious why not start this with 0?

Reply:
Mostly to avoid what will always be a match on `bos` token, however the prompt template/system prompt will still match so 1 isn't enough – I'll revert this to 0 👍